examples/tests/sdk_tests/bucket/events.w

```diff
@@ -1,60 +1,92 @@
 bring cloud;
 
-let b = new cloud.Bucket() as "b";
-let counter = new cloud.Counter();
+let b = new cloud.Bucket();
+let idsCounter = new cloud.Counter();
+let table = new cloud.Table(cloud.TableProps {
+  name: "key-history",
+  primaryKey: "_id",
+  columns: {
+    _id: cloud.ColumnType.STRING,
+    key: cloud.ColumnType.STRING,
+    operation: cloud.ColumnType.STRING,
+    source: cloud.ColumnType.NUMBER,
+    }
+  });
+
+  class Util {
+    extern "../external/sleep.js" static inflight sleep(milli: num);
+  }
+
+// the source of onDelete, onUpdate, onCreate are marked as: 1, and for onEvent it's marked as source 2.
+let logHistory = inflight (key: str, operation: str, source: num) => {
+  table.insert("${idsCounter.inc()}", Json { key: key, operation: operation, source: source  });
+};
+
 
 
 b.onDelete(inflight (key: str) => {
-    counter.inc();
+  logHistory(key, "DELETE", 1);
 });
 
 b.onUpdate(inflight (key: str) => {
-    counter.inc();
+  logHistory(key, "UPDATE", 1);
 });
 
-b.onCreate(inflight (key:str) => {
-    counter.inc();
+b.onCreate(inflight (key: str) => {
+  logHistory(key, "CREATE", 1);
 });
 
-b.onEvent(inflight (key: str) => {   
-    counter.inc();
+b.onEvent(inflight (key: str, event: cloud.BucketEventType) => { 
+  logHistory(key, "${event}", 2);
 });
 
 
-new cloud.Test(inflight () => {
+new cloud.Test(inflight () => {  
+  let wait = inflight (pred: inflight (): bool): bool => {
+    let var i = 0;
+    // waiting for up to 2 minutess, checking every 10 seconds
+    while i < 12 {
+      if pred() {
+        return true;
+      } 
+  
+      Util.sleep(10000);
+      i = i + 1;
+    }
+
+    return false;
+  };
 
-    inflight class Predicate {
-        counterVal: num;
-        init(counterVal: num) {
-            this.counterVal = counterVal;
+  let checkHitCount = inflight (key: str, operation: str, source: num, expectedVal: num): inflight (): bool => {
```

Comment:
This function has several arguments, so when someone reads the usage, it might not be apparent what each field means:

```js
checkHitCount("a", "CREATE", 1, 1);
```

This would be a great use case for a trailing struct argument! (I'm not sure if that's the official name of the feature but it's close enough.) You can specify a struct as the last argument, and then values of the struct can be inlined as keyword arguments to the function call, so it's easier to read, and the arguments can go in any order.

```js
struct CheckHitCountOptions {
  type: str;
  key: str;
  source: str;
  count: num;
}

let checkHitCount = inflight (opts: CheckHitCountOptions) {
  // snip
};

checkHitCount(type: "CREATE", key: "a", source: "ON_EVENT", count: 1)
```